### PR TITLE
Add cli options to write profiler output to InfluxDB

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

This patch adds the command line options and associated code to write the output of running the profiler to an InfluxDB instance, so we can create graphs like [1] with Grafana.

This is part of the work required to record and watch PWM results during CI to catch performance regressions.

[1] https://screenshots.firefox.com/j6sSZrN7pTuPK2kX/localhost

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17536)

### CODE REVIEW

File: components/config/opts.rs [hunk 1]
```diff
@@ -39,8 +39,13 @@ pub struct Opts {
...
     pub device_pixels_per_px: Option<f32>,
 
-    /// `None` to disable the time profiler or `Some` with an interval in seconds to enable it and
-    /// cause it to produce output on that interval (`-p`).
+    /// `None` to disable the time profiler or `Some` to enable it with:
+    ///  - an interval in seconds to cause it to produce output on that interval.
+    ///    (`i.e. -p 5`).
+    ///  - a file path to write profiling info to a TSV file upon Servo's termination.
+    ///    (`i.e. -p out.tsv`).
+    ///  - an InfluxDB hostname to store profiling info upon's Servo's termination.
```
Comment (jdm) [inline components/config/opts.rs]: upon

File: components/profile/time.rs [hunk 2]
```diff
@@ -391,7 +395,63 @@ impl Profiler {
...
+                    None => "",
+                };
+                let password = match *password {
+                    Some(ref p) => p,
+                    None => "",
+                };
+                let database = match *dbname {
+                    Some(ref n) => n,
+                    None => "",
+                };
```
Comment (jdm) [inline components/profile/time.rs]:
```rust
let user = user.as_ref().unwrap_or("");
let password = password.as_ref().unwrap_or("");
let database = dbname.as_ref().unwrap_or("");
```

File: components/profile/time.rs [hunk 3]
```diff
@@ -391,7 +395,63 @@ impl Profiler {
...
+                    username: user,
+                    password: password,
+                    database: database,
+                };
+
+                let hosts = vec![hostname.as_str()];
+                let client = create_client(credentials, hosts);
+
+                for (&(ref category, ref meta), ref mut data) in &mut self.buckets {
+                    data.sort_by(|a, b| {
```
Comment (jdm) [inline components/profile/time.rs]: I think you can use `data.sort()` instead.

File: components/profile/time.rs [hunk 4]
```diff
@@ -391,7 +395,63 @@ impl Profiler {
...
+                        measurement.add_field("mean", Value::Float(mean));
+                        measurement.add_field("median", Value::Float(median));
+                        measurement.add_field("min", Value::Float(min));
+                        measurement.add_field("max", Value::Float(max));
+                        if let Some(ref meta) = *meta {
+                            measurement.add_tag("host", meta.url.as_str());
+                        };
+                        match client.write_one(measurement, None) {
+                            Ok(_) => {},
+                            Err(_) => warn!("Could not write measurement to profiler db"),
```
Comment (jdm) [inline components/profile/time.rs]:
```rust
if client.write_one(...).is_err() {
    ...
}
```

**Review by jdm — CHANGES_REQUESTED** [review]
If it's useful to put the profiler data into a database, then this looks like a fine way to do it.

File: components/profile/time.rs (hunk 3 above)
Comment (ferjm) [inline components/profile/time.rs]: The trait `std::cmp::Ord` is not implemented for `f64`

**Pull request merged**